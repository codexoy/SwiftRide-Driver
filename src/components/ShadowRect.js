import React from "react"
import { View, StyleSheet } from "react-native"

export default class ShadowRect extends React.Component {
	render() {
		return (
			<View style={{ ...this.props.style, ...styles.container }}>
				{this.props.children}
			</View>
		)
	}
}

const styles = StyleSheet.create({
	container: {
		flex: 1,
		alignItems: "flex-start",
		backgroundColor: "#fff",
		shadowColor: "#000",
		shadowOffset: {
			width: 0,
			height: 6
		},
		shadowOpacity: 0.09,
		shadowRadius: 20,
		elevation: 6,
		marginLeft: 26,
		marginRight: 26,
		borderRadius: 4,
		padding: 20
	}
})
