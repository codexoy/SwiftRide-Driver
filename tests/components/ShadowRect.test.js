import { Text } from "react-native"
import React from "react"

import renderer from "react-test-renderer"

import ShadowRect from "../../src/components/ShadowRect"

describe("ShadowRect", () => {
	test("view renders correctly", () => {
		const tree = renderer
			.create(
				<ShadowRect style={{ minHeight: 320 }}>
					<Text>test</Text>
				</ShadowRect>
			)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})
})
