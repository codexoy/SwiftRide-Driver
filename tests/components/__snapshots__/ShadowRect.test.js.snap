// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShadowRect view renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderRadius": 4,
      "elevation": 6,
      "flex": 1,
      "marginLeft": 26,
      "marginRight": 26,
      "minHeight": 320,
      "padding": 20,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 6,
        "width": 0,
      },
      "shadowOpacity": 0.09,
      "shadowRadius": 20,
    }
  }
>
  <Text>
    test
  </Text>
</View>
`;
